lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,272 @@
+        self.payload["start"] = "2020-12"
+        self.payload["end"] = "2020-19"
+        response = client.post(
+            server_config.rest_uri + self.pbench_endpoint,
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.payload,
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert (
+            response.json.get("message")
+            == "Value '2020-19' (str) cannot be parsed as a date/time string"
+        )
+
+    def test_empty_query(
+        self,
+        client,
+        server_config,
+        query_api,
+        user_ok,
+        find_template,
+        build_auth_header,
+    ):
+        """
+        Check proper handling of a query resulting in no Elasticsearch matches.
+        PyTest will run this test multiple times with different values of the build_auth_header
+        fixture.
+        """
+        if not self.empty_es_response_payload or not self.elastic_endpoint:
+            pytest.skip("skipping the empty query test")
+        expected_status = HTTPStatus.OK
+        if build_auth_header["header_param"] != "valid":
+            expected_status = HTTPStatus.FORBIDDEN
+
+        index = self.build_index(server_config, ("2020-08", "2020-09", "2020-10"))
+        response = query_api(
+            self.pbench_endpoint,
+            self.elastic_endpoint,
+            self.payload,
+            index,
+            expected_status,
+            headers=build_auth_header["header"],
+            status=HTTPStatus.OK,
+            json=self.empty_es_response_payload,
+        )
+        assert response.status_code == expected_status
+        if response.status_code == HTTPStatus.OK:
+            assert response.json == []
+
+    @pytest.mark.parametrize(
+        "exceptions",
+        (
+            {
+                "exception": requests.exceptions.ConnectionError(),
+                "status": HTTPStatus.BAD_GATEWAY,
+            },
+            {
+                "exception": requests.exceptions.Timeout(),
+                "status": HTTPStatus.GATEWAY_TIMEOUT,
+            },
+            {
+                "exception": requests.exceptions.InvalidURL(),
+                "status": HTTPStatus.INTERNAL_SERVER_ERROR,
+            },
+            {"exception": Exception(), "status": HTTPStatus.INTERNAL_SERVER_ERROR},
+            {"exception": ValueError(), "status": HTTPStatus.INTERNAL_SERVER_ERROR},
+        ),
+    )
+    def test_http_exception(
+        self,
+        server_config,
+        query_api,
+        exceptions,
+        user_ok,
+        find_template,
+        pbench_token,
+    ):
+        """
+        Check that an exception in calling Elasticsearch is reported correctly.
+        """
+        if not self.elastic_endpoint:
+            pytest.skip("skipping the http exception test")
+
+        # Make the start and end time in payload the same to result in an exception
```

Comment:
Huh? We're going to force an exception on the Elasticsearch calls; we just need *valid* dates to get through the parameter parsing before we get to that point. (And even with your mock matching issue, which is reasonable enough although we could always build the expected URL to match the actual dates, the identical dates are only *enabling* not *causing* the exception, by ensuring that the mock matches the actual URL.)

In fact, this still requires coordination between setting up the mock address and construction of the query, so it would be better to do both consistently here, in which case the date values don't matter. (Although a URL with a single index name is easier/faster to construct than 15 separate index names!)

Reply:
I really don't like these unclear comments, and I wish you'd fixed them. I'm going to approve anyway with slight reluctance because I'd like to move past this. But if you make more changes for other reasons, please take a look at this again.

Reply:
Fixed. Yeah, this code should't be here.